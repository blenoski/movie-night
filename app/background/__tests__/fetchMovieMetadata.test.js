'use strict'
/* globals describe, test, expect, jest */

// Mock the OMDb API responses.
jest.mock('../omdb', () => {
  return {
    fetchMovieMetadata: jest.fn(movieFile => {
      if (movieFile.includes('reject')) {
        return Promise.reject(new Error())
      }

      const url = movieFile
      let metadata = {}
      if (movieFile.includes('imgUrl')) {
        metadata.imgUrl = 'http://poster.png'
        metadata.imdbID = 'tt123'
      }
      return Promise.resolve({ metadata, url })
    })
  }
})

// Mock the poster image path returned from config so snapshots
// do not contain any host information.
jest.mock('../../../config', () => {
  return {
    posterImagePath: '/path/to/movie/posters'
  }
})

// This is the module we are testing
const { fetchMovieMetadata } = require('../fetchMovieMetadata')

describe('fetchMovieMetadata', () => {
  test('resolves with fileInfo metadata on successful fetch', () => {
    const movieFile = 'OldSchool.avi'
    return expect(fetchMovieMetadata(movieFile))
      .resolves.toMatchSnapshot()
  })

  test('resolves with imgFile when metadata contains imgUrl', () => {
    const movieFile = 'OldSchool imgUrl'
    return expect(fetchMovieMetadata(movieFile))
      .resolves.toMatchSnapshot()
  })

  test('rejects on omdb fetch error', () => {
    const movieFile = 'reject'
    return expect(fetchMovieMetadata(movieFile))
      .rejects.toBeInstanceOf(Error)
  })
})
