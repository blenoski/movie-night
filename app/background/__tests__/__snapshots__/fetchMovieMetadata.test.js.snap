// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`fetchMovieMetadata resolves with fileInfo metadata on successful fetch 1`] = `
Object {
  "fileInfo": Array [
    Object {
      "location": "OldSchool.avi",
      "query": "OldSchool.avi",
    },
  ],
  "imgFile": "",
}
`;

exports[`fetchMovieMetadata resolves with imgFile when metadata contains imgUrl 1`] = `
Object {
  "fileInfo": Array [
    Object {
      "location": "OldSchool imgUrl",
      "query": "OldSchool imgUrl",
    },
  ],
  "imdbID": "tt123",
  "imgFile": "/Users/blenoski/Developer/ConfidentCruiser/movie-night/appdata/image/tt123.png",
  "imgUrl": "http://poster.png",
}
`;
